### PR TITLE
feat: support docker image creation from local codebase

### PR DESCRIPTION
**Pull request by shreemaan-abhishek**
Transferred from https://github.com/apache/apisix/pull/9586

### CODE REVIEW

**Comment by lingsamuel** [comment]
ci failed

**Comment by shreemaan-abhishek** [comment]
@lingsamuel, please re-run the CI. The error shown is unrelated.

**Comment by lingsamuel** [comment]
still failing 🤔

**Comment by shreemaan-abhishek** [comment]
This CI has been failing everywhere consistently:

https://github.com/apache/apisix-docker/actions/workflows/apisix-docker-example-test.yaml

**Comment by lingsamuel** [comment]
Re-ran several times but still failed 🥺 cc @AlinsRan

**Comment by AlinsRan** [comment]
> Re-ran several times but still failed 🥺 cc @AlinsRan

Let me check it.

**Comment by AlinsRan** [comment]
Re-uploaded. Let's continue.

**Comment by shreemaan-abhishek** [comment]
@lingsamuel CI passing.

**Review by Sn0rt — APPROVED** [review]

**Comment by Sn0rt** [comment]
@AlinsRan pls help me to review the PR

**Comment by AlinsRan** [comment]
We should increase ci to ensure that the locally built image can work.

File: .github/workflows/apisix_dev_docker_local_test.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
```
Comment (AlinsRan) [inline .github/workflows/apisix_dev_docker_local_test.yaml:27]: This is an image built on Debian, why does it need to run on multiple platforms?

Comment (shreemaan-abhishek) [reply .github/workflows/apisix_dev_docker_local_test.yaml:27]: Makes sense, I just pushed my changes.

**Review by AlinsRan — APPROVED** [review]

File: debian-dev/Dockerfile.local [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+RUN set -x \
+    && (test "${ENABLE_PROXY}" != "true" || /bin/sed -i 's,http://deb.debian.org,http://mirrors.aliyun.com,g' /etc/apt/sources.list) \
+    && apt-get -y update --fix-missing \
+    && apt-get install -y curl \
+        gawk \
+        git \
+        libldap2-dev \
+        liblua5.1-0-dev \
+        lua5.1 \
```
[the remark this answers is not in the document]
Comment (shreemaan-abhishek) [reply debian-dev/Dockerfile.local:47]: without lua interpreter, the build would fail:

image

**Review by leslie-tsang — APPROVED** [review]

**Pull request merged**